x/group/server/msg_server.go

```diff
@@ -0,0 +1,599 @@
+				if err != nil {
+					return err
+				}
+				err = math.SafeSub(totalWeight, totalWeight, previousMemberWeight)
+				if err != nil {
+					return err
+				}
+				if err := s.groupMemberTable.Save(ctx, &groupMember); err != nil {
+					return sdkerrors.Wrap(err, "add member")
+				}
+			} else if err := s.groupMemberTable.Create(ctx, &groupMember); err != nil {
+				return sdkerrors.Wrap(err, "add member")
+			}
+			err = math.Add(totalWeight, totalWeight, newMemberWeight)
+			if err != nil {
+				return err
+			}
+			g.TotalWeight = math.DecimalString(totalWeight)
+		}
+		g.Version++
+		return s.groupTable.Save(ctx, g.GroupId.Bytes(), g)
+	}
+
+	err := s.doUpdateGroup(ctx, req, action, "members updated")
+	if err != nil {
+		return nil, err
+	}
+
+	return &group.MsgUpdateGroupMembersResponse{}, nil
+}
+
+func (s serverImpl) UpdateGroupAdmin(ctx types.Context, req *group.MsgUpdateGroupAdminRequest) (*group.MsgUpdateGroupAdminResponse, error) {
+	action := func(g *group.GroupInfo) error {
+		g.Admin = req.NewAdmin
+		g.Version++
+		return s.groupTable.Save(ctx, g.GroupId.Bytes(), g)
+	}
+
+	err := s.doUpdateGroup(ctx, req, action, "admin updated")
+	if err != nil {
+		return nil, err
+	}
+
+	return &group.MsgUpdateGroupAdminResponse{}, nil
+}
+
+func (s serverImpl) UpdateGroupMetadata(ctx types.Context, req *group.MsgUpdateGroupMetadataRequest) (*group.MsgUpdateGroupMetadataResponse, error) {
+	action := func(g *group.GroupInfo) error {
+		g.Metadata = req.Metadata
+		g.Version++
+		return s.groupTable.Save(ctx, g.GroupId.Bytes(), g)
+	}
+
+	err := s.doUpdateGroup(ctx, req, action, "metadata updated")
+	if err != nil {
+		return nil, err
+	}
+
+	return &group.MsgUpdateGroupMetadataResponse{}, nil
+}
+
+func (s serverImpl) CreateGroupAccount(ctx types.Context, req *group.MsgCreateGroupAccountRequest) (*group.MsgCreateGroupAccountResponse, error) {
+	admin, err := sdk.AccAddressFromBech32(req.GetAdmin())
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "request admin")
+	}
+	policy := req.GetDecisionPolicy()
+	groupID := req.GetGroupID()
+	metadata := req.GetMetadata()
+
+	if err := assertMetadataLength(metadata, s.maxMetadataLength(ctx), "group account metadata"); err != nil {
+		return nil, err
+	}
+
+	g, err := s.getGroupInfo(ctx, groupID)
+	if err != nil {
+		return nil, err
+	}
+	groupAdmin, err := sdk.AccAddressFromBech32(g.Admin)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "group admin")
+	}
+	if !groupAdmin.Equals(admin) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "not group admin")
+	}
+	accountAddr := group.AccountCondition(s.groupAccountSeq.NextVal(ctx)).Address()
+	groupAccount, err := group.NewGroupAccountInfo(
+		accountAddr,
+		groupID,
+		admin,
+		metadata,
+		1,
+		policy,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.groupAccountTable.Create(ctx, &groupAccount); err != nil {
+		return nil, sdkerrors.Wrap(err, "could not create group account")
+	}
```

Comment:
We also need to make sure that there is a module account created in `x/auth`. When we update with ADR 028 let's do this. Can you add a TODO @blushi ?

Reply:
added